### PR TITLE
[ML] Enable OpenAI Streaming

### PR DESCRIPTION
**Pull request by prwhelan**
Enables end-to-end streaming for OpenAI's chat completion API.

Additional changes:
- InferenceServiceResults now returns a wildcard, allowing for tests to work with classes that extend ChunkedToXContent.
- StreamingChatCompletionResults now defines the Results structure so that future providers can reuse the same structure.
- DelegatingProcessor now cancels the upstream if `next` throws an exception and forwards the exception downstream.
- Moved the streaming handler from OpenAiChatCompletionResponseHandler to OpenAiResponseHandler so that Azure Open AI can reuse it.
- OpenAiStreamingProcessor now iterates over the returned choices array, handling both OpenAI and Azure response formats.
- SenderService declares a helper `Set<TaskType>` for implementations to reuse when enabling streaming.
- Added an InferenceEventsAssertion test helper to fluidly assert responses from the mock webserver.

### CODE REVIEW

**Comment by prwhelan** [comment]
@elasticmachine update branch

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/InferenceEventsAssertion.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+import static org.elasticsearch.xcontent.ToXContent.EMPTY_PARAMS;
+import static org.hamcrest.CoreMatchers.is;
+
+/**
+ * Helper to chain together assertions for streaming {@link InferenceServiceResults}.
+ */
+public record InferenceEventsAssertion(Iterator<String> events, Throwable error, boolean isComplete, int iterations) {
+
+    public static InferenceEventsAssertion assertThat(InferenceServiceResults results) throws Exception {
```
Comment (davidkyle) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/InferenceEventsAssertion.java:37]: Nice

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by prwhelan** [comment]
@elasticmachine update branch

**Pull request merged**